src/main/java/appeng/core/AELog.java

```diff
@@ -19,93 +19,208 @@
 	{
 		if( AEConfig.instance == null || AEConfig.instance.isFeatureEnabled( AEFeature.Logging ) )
 		{
-			FMLRelaunchLog.log( "AE2:" + ( Platform.isServer() ? "S" : "C" ), level, format, data );
+			final String formattedMessage = String.format( message, params );
+			final Logger logger = getLogger();
+
+			logger.log( level, formattedMessage );
 		}
 	}
 
-	public static void grinder( final String o )
+	/**
+	 * Log an exception with a custom message formated via {@link String#format(String, Object...)}
+	 *
+	 * Similar to {@link AELog#log(Level, String, Object...)}.
+	 *
+	 * @see AELog#log(Level, String, Object...)
+	 *
+	 * @param exception
+	 * @param message the message to be formatted.
+	 * @param params the parameters used for {@link String#format(String, Object...)}.
+	 */
+	public static void log( @Nonnull final Throwable exception, @Nonnull String message, final Object... params )
 	{
-		if( AEConfig.instance.isFeatureEnabled( AEFeature.GrinderLogging ) )
+		if( AEConfig.instance.isFeatureEnabled( AEFeature.Logging ) )
 		{
-			log( Level.DEBUG, "grinder: " + o );
+			final String formattedMessage = String.format( message, params );
+			final Logger logger = getLogger();
+
+			logger.log( Level.WARN, formattedMessage, exception );
 		}
 	}
 
-	public static void integration( final Throwable exception )
+	/**
+	 * @see AELog#log(Level, String, Object...)
+	 * @param format
+	 * @param params
+	 */
+	public static void info( @Nonnull final String format, final Object... params )
 	{
-		if( AEConfig.instance.isFeatureEnabled( AEFeature.IntegrationLogging ) )
-		{
-			error( exception );
-		}
+		log( Level.INFO, format, params );
 	}
 
-	public static void error( final Throwable e )
+	/**
+	 * @see AELog#log(Level, String, Object...)
+	 * @param format
+	 * @param params
+	 */
+	public static void warning( @Nonnull final String format, final Object... params )
 	{
-		if( AEConfig.instance.isFeatureEnabled( AEFeature.Logging ) )
+		log( Level.WARN, format, params );
+	}
+
+	/**
+	 * @see AELog#log(Level, String, Object...)
+	 * @param format
+	 * @param params
+	 */
+	public static void error( @Nonnull final String format, final Object... params )
+	{
+		log( Level.ERROR, format, params );
+	}
+
+	/**
+	 * @see AELog#log(Level, String, Object...)
+	 * @param format
+	 * @param data
+	 */
+	public static void debug( @Nonnull final String format, final Object... data )
+	{
+		if( AEConfig.instance.isFeatureEnabled( AEFeature.DebugLogging ) )
 		{
-			severe( "Error: " + e.getClass().getName() + " : " + e.getMessage() );
-			e.printStackTrace();
+			log( Level.DEBUG, format, data );
 		}
 	}
 
-	public static void severe( final String format, final Object... data )
+	/**
+	 * Use this to log caught {@link Exception} inside a try/catch block when necessary.
+	 *
+	 * Mostly intended for ignored exceptions.
+	 *
+	 * @param exception
+	 */
+	public static void catching( @Nonnull final Throwable exception )
```

Comment:
I dont think the intention is clear enough. I would rather not log the exception before logging this for non purpose. Either its a warning, bug or additional information. I would not know how to fit in this usecase.
Because why are all errors suddenly moved to WARN? An error is clearly above a warning level.

The problem arises from the unclear uses of exceptions. Obviously the codebase is too complex to just remove these.